src/coreComponents/physicsSolvers/multiphysics/PoromechanicsSolver.hpp

```diff
@@ -0,0 +1,270 @@
+    }
+    return result;
+  }
+
+  real64 computeAitkenRelaxationFactor( const array1d< real64 > & s0,
+                                        const array1d< real64 > & s1,
+                                        const array1d< real64 > & s1_tilde,
+                                        const array1d< real64 > & s2_tilde,
+                                        const real64 omega0 )
+  {
+    array1d< real64 > r1 = addTwoVecs( s1_tilde, s0, -1.0 );
+    array1d< real64 > r2 = addTwoVecs( s2_tilde, s1, -1.0 );
+
+    // diff = r2 - r1
+    array1d< real64 > diff = addTwoVecs( r2, r1, -1.0 );
+
+    const real64 denom = dotTwoVecs( diff, diff );
+    const real64 numer = dotTwoVecs( r1, diff );
+
+    real64 omega1 = 1.0;
+    if( !isZero( denom ))
+    {
+      omega1 = -1.0 * omega0 * numer / denom;
+    }
+    return omega1;
+  }
+
+  array1d< real64 > computeUpdate( const array1d< real64 > & s1,
+                                   const array1d< real64 > & s2_tilde,
+                                   const real64 omega1 )
+  {
+    return addTwoVecs( scalarMultiplyAVec( s1, 1.0 - omega1 ),
+                       scalarMultiplyAVec( s2_tilde, omega1 ),
+                       1.0 );
+  }
+
+  void startSequentialIteration( integer const & iter,
+                                 DomainPartition & domain ) override
+  {
+    if( m_useNA )
+    {
+      if( iter == 0 )
+      {
+        recordAverageMeanTotalStressIncrement( domain, m_s1 );
+      }
+      else
+      {
+        m_s0 = m_s1;
+        m_s1 = m_s2;
+        m_s1_tilde = m_s2_tilde;
+        m_omega0 = m_omega1;
+      }
+    }
+  }
+
+  void finishSequentialIteration( integer const & iter,
+                                  DomainPartition & domain ) override
+  {
+    if( m_useNA )
+    {
+      if( iter == 0 )
+      {
+        m_s2 = m_s2_tilde;
+        m_omega1 = 1.0;
+      }
+      else
+      {
+        m_omega1 = computeAitkenRelaxationFactor( m_s0, m_s1, m_s1_tilde, m_s2_tilde, m_omega0 );
+        m_s2 = computeUpdate( m_s1, m_s2_tilde, m_omega1 );
+        applyAcceleratedAverageMeanTotalStressIncrement( domain, m_s2 );
+      }
+    }
+  }
+
+  virtual void mapSolutionBetweenSolvers( DomainPartition & domain, integer const solverType ) override
+  {
+    if( m_useNA && solverType == static_cast< integer >( SolverType::SolidMechanics ) )
+    {
+      recordAverageMeanTotalStressIncrement( domain, m_s2_tilde );
+    }
+  }
+
+  virtual void
+  postProcessInput() override
+  {
+    Base::postProcessInput();
+
+    if( m_useNA && MpiWrapper::commSize( MPI_COMM_GEOSX ) > 1 )
+    {
+      GEOS_ERROR( "Nonlinear acceleration is not implemented for MPI runs" );
+    }
+  }
+
+  /// member variables needed for Nonlinear Acceleration (Aitken)
+  integer m_useNA;
+  array1d< real64 > m_s0;
+  array1d< real64 > m_s1;
+  array1d< real64 > m_s1_tilde;
+  array1d< real64 > m_s2;
+  array1d< real64 > m_s2_tilde;
+  real64 m_omega0;
+  real64 m_omega1;
```

Comment:
Maybe use more meaningful names if these variables have a physical meaning, or add some lines to comment on their future use